code/modules/projectiles/projectile/tesla.dm

```diff
@@ -0,0 +1,125 @@
+#define MAX_LIGHTNING_PER_PULSE 3
+#define LIGHTNING_RANGE 3
+#define PULSE_SHOCK_CHANCE 50
+#define MIN_LIGHTNING_DAMAGE 10
+
+/obj/item/projectile/teslaball
+	name = "tesla ball"
+	icon = 'icons/obj/projectiles_experimental.dmi'
+	icon_state = "teslaball_normal"
+	damage = 0
+	fire_sound = 'sound/weapons/wave.ogg'
+	phase_type = PROJREACT_WINDOWS|PROJREACT_OBJS|PROJREACT_MOBS
+	penetration = -1
+	kill_count = 30
+	projectile_speed = 3
+
+	var/yellow = 0
+	var/charge = 0
+	var/list/shocked_people = list()    //each person only gets shocked once
+	
+
+/obj/item/projectile/teslaball/New()
+	..()
+	spawn(1.3 SECONDS)  
+		Pulse()
+
+/obj/item/projectile/teslaball/to_bump(atom/A)
+	..()
+	Zap(A, FALSE)
+
+/obj/item/projectile/teslaball/OnDeath()
+	for(var/mob/living/M in view(LIGHTNING_RANGE,src))
```

Comment:
I notice this doesn't respect shocked_people, meaning they could get zapped twice. Intended?

Reply:
I completely forgot about the shocked_people list, its unused. Each person can get shocked once per pulse, so getting zapped more than once is possible.